File: agents/kos_repo_analysis_claude/kos_repo_analysis_export/code/klf_core/router.ts
import { KLFEnvelope } from "./types";
import { getHandler } from "./intent-registry";
import { InvalidEnvelopeError, RoutingError } from "./errors";
import { isValidEnvelope } from "./envelope";

export async function routeMessage(envelope: KLFEnvelope): Promise<any> {
  if (!isValidEnvelope(envelope)) {
    throw new InvalidEnvelopeError();
  }

  const handler = getHandler(envelope.intent);
  if (!handler) {
    throw new RoutingError(`No handler registered for intent "${envelope.intent}"`);
  }

  return await handler(envelope.payload, envelope.meta);
} 

File: agents/kos_repo_analysis_claude/kos_repo_analysis_export/code/backend_gateway/klf/router.py
from klf.intent_registry import get_handler
from klf.types import KLFEnvelope
from klf.errors import InvalidEnvelopeError

async def route_message(envelope: KLFEnvelope):
    if not envelope.intent or not envelope.from_ or not envelope.to:
        raise InvalidEnvelopeError("Missing required envelope fields")

    handler = get_handler(envelope.intent)
    if not handler:
        raise Exception(f"No handler for intent {envelope.intent}")

    return await handler(envelope.payload, envelope.meta or {}) 